crates/cubecl-wgpu/src/compute/server.rs

```diff
@@ -221,8 +221,21 @@ impl ComputeServer for WgpuServer {
     }
 
     fn read_tensor(&mut self, bindings: Vec<BindingWithMeta>) -> DynFut<Vec<Vec<u8>>> {
+        let expected_sizes = bindings
+            .iter()
+            .map(|it| it.shape.iter().product::<usize>() * it.elem_size)
+            .collect::<Vec<_>>();
         let bindings = bindings.into_iter().map(|it| it.binding).collect();
-        self.read(bindings)
+        let data = self.read(bindings);
+        Box::pin(async move {
+            let mut data = data.await;
+
+            for (data, expected_size) in data.iter_mut().zip(expected_sizes) {
+                data.truncate(expected_size);
```

Comment:
Why is there an expected size now?

Reply:
Because for tensors we actually know the size of the element, so we can properly truncate the read bytes. CUDA already does this because of how the function is implemented, so this just matches functionality. Previously, burn would need to handle this because CubeCL didn't know the element size or shape and couldn't tell when a handle was padded for alignment.